### ISSUE TITLE
Unable to pin a message in a newly created group

### ISSUE THREAD

**Issue by supolka** [tdlib/td#1182]
Steps to reproduce:
1. Create a new group
2. Send a message to it
3. Try to pin the message from 2

Expected result: ok, message is pinned

Actual result: error 6: Message can't be pinned

Message can be pinned from Telegram client on phone. Is this behavior intended and I am missing something? Thanks

**Comment by levlam** [comment]
Are you trying to pin a message before it is actually sent?
Can it be pinned after you close and reopen the chat?

**Comment by supolka** [comment]
Yes, that was indeed the problem. Thank you very much for the help, it wasn't obvious that I have to wait for update.

The sequence that worked was:
1. Send a message
2. Wait for update of type updateMessageSendSucceeded in that chat, note the message.id in that update
3. Use message.id from the update to pin the message

Quick note: while looking for workarounds I tried slightly different approach for chat where message has already been sent:
1. Get chat
2. Pin the message with id chat.LastMessage.id
This also didn't work. (I should note that I didn't had any loop to fetch the updates in the code). Apparently tdlib maintains a cache, and when I call `GetChat` the value from cache is returned where message is still temporary. Once I've added listening to the updates - it started to work.
